### PR TITLE
AI/TFLM : Add Tensor Flow Micro source code

### PR DESCRIPTION
**Pull request by vibhor-m**
1. Add build config for Tensor Flow Micro library
2. Add Tensor Flow Micro source code
3. Add Tensor Flow Micro Sample Application
4. AI Framework based on Tensor Flow Micro
5. Sample application for AI Framework usage

### CODE REVIEW

File: apps/examples/hello_tfmicro/Kconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+config EXAMPLES_HELLO_TFMICRO
+	bool "\"Hello, Tensorflow Lite Micro!\" C++ example"
+	default n
+	depends on HAVE_CXXINITIALIZE && EXTERNAL_TFMICRO
+	---help---
+		Enable the \"Hello, Tensorflow Lite Micro!\" C++ example
+
+config USER_ENTRYPOINT
+	string
+	default "hello_tfmicro_main" if ENTRY_HELLO_TFMICRO
```
Comment (sunghan-chang) [inline apps/examples/hello_tfmicro/Kconfig:15]: `ENTRY_HELLO_TFMICRO` is mismatched with Kconfig_ENTRY, `config ENTRY_TFMICRO`.
That's wrong. Could you check it?

File: apps/examples/hello_tfmicro/Make.defs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+###########################################################################
+#
+# Copyright 2020 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/hello_tfmicro/Make.defs:3]: New files should have Samsung Apache license with 2023.

File: apps/examples/hello_tfmicro/Makefile [hunk 3]
```diff
@@ -0,0 +1,143 @@
+###########################################################################
+#
+# Copyright 2020 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline apps/examples/hello_tfmicro/Makefile:3]: Please check all of newly added files which have Samsung license.

File: apps/examples/hello_tfmicro/main.cc [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+	// Output the results. A custom HandleOutput function can be implemented
+	// for each supported hardware target.
+	HandleOutput(error_reporter, x_val, y_val);
+
+	// Increment the inference_counter, and reset it if we have reached
+	// the total number per cycle
+	inference_count += 1;
+	if (inference_count >= kInferencesPerCycle)
+	{
```
Comment (sunghan-chang) [inline apps/examples/hello_tfmicro/main.cc:145]: Except external, arch and bsp, we should follow TizenRT coding rules. Could you fix violations of coding rules?
```
if ( )
{
->
if ( ) {
```
```
void xx()
{
->
void xx(void)
{
```
spaces -> tabs for indetation

File: apps/examples/hello_tfmicro/Makefile (hunk 3 above)
Comment (vibhor-m) [reply apps/examples/hello_tfmicro/Makefile]: sorry, previously i missed this instance.

File: external/tfmicro/tensorflow/lite/core/api/error_reporter.cc [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
```
Comment (amogh-samsung) [inline external/tfmicro/tensorflow/lite/core/api/error_reporter.cc:1]: Is there a need for Samsung electronics License addition in external files? Kindly check

Comment (vibhor-m) [reply external/tfmicro/tensorflow/lite/core/api/error_reporter.cc:1]: @Taejun-Kwon What is your opinion? There are many external files.

File: apps/examples/hello_tfmicro/constants.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (vibhor-m) [reply apps/examples/hello_tfmicro/constants.h:1]: added

File: apps/examples/hello_tfmicro/sine_model_data.h [hunk 7]
```diff
@@ -0,0 +1,27 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (vibhor-m) [reply apps/examples/hello_tfmicro/sine_model_data.h:1]: added

File: external/tfmicro/Kconfig [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+# see kconfig-language at https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt
+#
+
+config EXTERNAL_TFMICRO
+	bool "Tensorflow Lite for Microcontroller"
+	default n
+	select HAVE_CXX
+	select HAVE_CXXINITIALIZE
+	---help---
+		Enables the Tensorflow lite for Microcontroller
```
[the remark this answers is not in the document]
Comment (vibhor-m) [reply external/tfmicro/Kconfig:12]: done

File: apps/examples/lw_aifw_sine_test_app/lwaifw_sine_test_input.h [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+ * @brief Model input
+ */
+
+#ifndef __lwaifw_sine_test_input_h__
+#define __lwaifw_sine_test_input_h__
+
+#define NO_OF_ROWS_TO_READ 20
+#define NO_OF_SENSORS 1
+
+float gDataValues[NO_OF_ROWS_TO_READ][NO_OF_SENSORS] = {
```
Comment (Taejun-Kwon) [inline apps/examples/lw_aifw_sine_test_app/lwaifw_sine_test_input.h:30]: if you brought these value from tflite sample, you should put their license

Comment (vibhor-m) [reply apps/examples/lw_aifw_sine_test_app/lwaifw_sine_test_input.h:30]: hello_tflite application generates values at runtime, I copied these values from hello_tflite output, therefore license may not be required

**Review by Taejun-Kwon — APPROVED** [review]

**Pull request merged**